File: javapatterns/src/main/java/org/iut/nantes/factories/pizzas/factorymethod/Pizza.java
package org.iut.nantes.factories.pizzas.factorymethod;

abstract class Pizza {
    protected String fromage;
    protected String topping;
    protected String sauce;

    public void preparer() {
        System.out.println("Je prépare ma pizza");
    }

    public void cuire() {
        System.out.println("Je cuis ma pizza");
    }

    public void couper() {
        System.out.println("Je coupe ma pizza");
    }

    public void dansLaBoite() {
        System.out.println("Je mets ma pizza dans la boite");
    }
}

class RoyaleNante extends Pizza {
    public RoyaleNante() {
        fromage = "curé nantais";
        topping = "champignon de paris";
        sauce = "tomate Française";
    }
}

class RoyaleTokyo extends Pizza {
    public RoyaleTokyo() {
        fromage = "sakura chīzu";
        topping = "kikurage";
        sauce = "tomate";
    }
}

class RoyaleNY extends Pizza {
    public RoyaleNY() {
        fromage = "chedar";
        topping = "mushrooms";
        sauce = "tomate";
    }
}

class VegetarienneNantes extends Pizza {
    public VegetarienneNantes() {
        fromage = "curé nantais";
        topping = "tofu";
        sauce = "tomate";
    }
}

class VegetarienneTokyo extends Pizza {
    public VegetarienneTokyo() {
        fromage = "osef";
        topping = "tofu";
        sauce = "cream";
    }
}

class VegetarienneNY extends Pizza {
    public VegetarienneNY() {
        fromage = "chedar";
        topping = "olives";
        sauce = "cream";
    }
}

File: kotlinpatterns/src/main/kotlin/org/iut/nantes/factories/pizzas/factorymethod/Pizza.kt
package org.iut.nantes.factories.pizzas.factorymethod

abstract class Pizza {
    protected var fromage: String? = null
    protected var topping: String? = null
    protected var sauce: String? = null
    fun preparer() {
        println("Je prépare ma pizza")
    }

    fun cuire() {
        println("Je cuis ma pizza")
    }

    fun couper() {
        println("Je coupe ma pizza")
    }

    fun dansLaBoite() {
        println("Je mets ma pizza dans la boite")
    }
}

class RoyaleNante : Pizza() {
    init {
        fromage = "curé nantais"
        topping = "champignon de paris"
        sauce = "tomate Française"
    }
}

class RoyaleTokyo : Pizza() {
    init {
        fromage = "sakura chīzu"
        topping = "kikurage"
        sauce = "tomate"
    }
}

class RoyaleNY : Pizza() {
    init {
        fromage = "chedar"
        topping = "mushrooms"
        sauce = "tomate"
    }
}

class VegetarienneNantes : Pizza() {
    init {
        fromage = "curé nantais"
        topping = "tofu"
        sauce = "tomate"
    }
}

class VegetarienneTokyo : Pizza() {
    init {
        fromage = "osef"
        topping = "tofu"
        sauce = "cream"
    }
}

class VegetarienneNY : Pizza() {
    init {
        fromage = "chedar"
        topping = "olives"
        sauce = "cream"
    }
}